app/models/framework_response.rb

```diff
@@ -42,6 +42,13 @@ def update_with_flags!(value)
     raise ActiveRecord::ReadOnlyRecord, "Can't update framework_responses because person_escort_record is #{person_escort_record.status}"
   end
 
+  def value=(raw_value)
+    unless raw_value.blank? || value_type_valid?(raw_value)
```

Comment:
Feels weird to reach up and then reach down the inheritance hierarchy to do common then specific validations. I don't think it's unforgivable but certainly it's unexpected.

Would it be more obvious to just duplicate the common blank value check and error adding, here?

Reply:
I agree this isn't the best approach, I tried going through the before validation callback initially, but the problem is that we need this to fail before trying to set the value, due to the nested structures and the way they are built. Happy to discuss other ways of failing early.
Not sure I understand what you mean with duplicating it, as in move this to each STI class? I think it would be nice to utilise the base class to avoid duplication for type validations

Reply:
Yeah, duplicate in the STI class was my thinking. It's okay